src/__tests__/actions-test.ts

```diff
@@ -30,92 +31,92 @@ describe("actions", () => {
+
       expect(dispatch.callCount).to.equal(3);
       expect(dispatch.args[0][0].type).to.equal(`${type}_${ActionCreator.REQUEST}`);
       expect(dispatch.args[1][0].type).to.equal(`${type}_${ActionCreator.SUCCESS}`);
       expect(dispatch.args[2][0].type).to.equal(`${type}_${ActionCreator.LOAD}`);
       expect(dispatch.args[2][0].data).to.equal("response");
-      expect(fetchMock.callCount).to.equal(1);
-      expect(fetchMock.args[0][0]).to.equal(url);
-      expect(fetchMock.args[0][1].method).to.equal("POST");
+      expect(fetchMock.called()).to.equal(true);
+      expect(fetchArgs[0][0]).to.equal(url);
+      expect(fetchArgs[0][1].method).to.equal("POST");
       const expectedHeaders = new Headers();
-      expect(fetchMock.args[0][1].headers).to.deep.equal(expectedHeaders);
-      expect(fetchMock.args[0][1].body).to.equal(formData);
+      expect(fetchArgs[0][1].headers).to.deep.equal(expectedHeaders);
+      expect(fetchArgs[0][1].body).to.equal(formData);
     });
 
     it("postForm with JSON response dispatches request, success, and load", async () => {
       const dispatch = stub();
       const textResponse = "{\"id\": \"test\", \"name\": \"test\"}";
-      const responseText = stub().returns(new Promise<any>((resolve) => {
-        resolve(textResponse);
-      }));
-      const fetchMock = stub().returns(new Promise<any>((resolve, reject) => {
-        resolve({ status: 200, text: responseText });
-      }));
-      fetch = fetchMock;
+      fetchMock
+        .mock(url, { status: 200, text: textResponse });
 
       await actions.postForm(type, url, formData, "POST", "", "JSON")(dispatch);
+      let fetchArgs = fetchMock.calls();
+
       expect(dispatch.callCount).to.equal(3);
       expect(dispatch.args[0][0].type).to.equal(`${type}_${ActionCreator.REQUEST}`);
       expect(dispatch.args[1][0].type).to.equal(`${type}_${ActionCreator.SUCCESS}`);
       expect(dispatch.args[2][0].type).to.equal(`${type}_${ActionCreator.LOAD}`);
-      expect(dispatch.args[2][0].data).to.equal(textResponse);
-      expect(fetchMock.callCount).to.equal(1);
-      expect(fetchMock.args[0][0]).to.equal(url);
-      expect(fetchMock.args[0][1].method).to.equal("POST");
+      expect(dispatch.args[2][0].data.text).to.equal(textResponse);
+      expect(fetchMock.called()).to.equal(true);
+      expect(fetchArgs[0][0]).to.equal(url);
+      expect(fetchArgs[0][1].method).to.equal("POST");
       const expectedHeaders = new Headers();
-      expect(fetchMock.args[0][1].headers).to.deep.equal(expectedHeaders);
-      expect(fetchMock.args[0][1].body).to.equal(formData);
+      expect(fetchArgs[0][1].headers).to.deep.equal(expectedHeaders);
+      expect(fetchArgs[0][1].body).to.equal(formData);
     });
 
     it("dispatches a DELETE request", async () => {
       const dispatch = stub();
-      const responseText = stub().returns(new Promise<string>((resolve) => {
+      const responseText = new Promise<string>((resolve) => {
```

Comment:
Why is this a promise when the previous one is just text and the one before that is a stub? They were more similar before.

Reply:
I was testing all versions out and forgot to just choose one. Will update.